### PR TITLE
Logged-in Performance Profiler: Show recommendations and filter by selected recommendation

### PR DESCRIPTION
**Pull request by vykes-mac**
Closes https://github.com/Automattic/dotcom-forge/issues/9177

## Proposed Changes

* Display the number of recommended fixes on each metric
* Allow filtering recommendations based on selected metric

image

## Why are these changes being made?

*

## Testing Instructions

* Go to `/site-performance/site`
* Select a metric and click `view recommendations`
* Verify it scroll to recommendations section and it's filtered by the selected metric

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/hosting/performance/site-performance.tsx [hunk 1]
```diff
@@ -131,6 +133,19 @@ export const SitePerformance = () => {
...
+		setRecommendationsFilter( filter );
+		const url = new URL( window.location.href );
+
+		if ( filter ) {
+			url.searchParams.set( 'filter', filter );
+		} else {
+			url.searchParams.delete( 'filter' );
+		}
+
+		window.history.replaceState( {}, '', url.toString() );
```
Comment (zaguiini) [inline client/hosting/performance/site-performance.tsx:146]: We can do `page.replace` like we're doing in the `PageSelector` prop.

File: client/performance-profiler/components/status-section/style.scss [hunk 2]
```diff
@@ -9,7 +9,7 @@
 
 	.status-badge {
 		display: flex;
-		padding: 6px 0;
```
Comment (zaguiini) [inline client/performance-profiler/components/status-section/style.scss:12]: Will this change affect the logged-out version?

Comment (vykes-mac) [reply client/performance-profiler/components/status-section/style.scss:12]: That component is unique to only logged in version

File: client/hosting/performance/site-performance.tsx (hunk 1 above)
Comment (vykes-mac) [reply client/hosting/performance/site-performance.tsx:146]: I tried page.replace but it somehow affects the scroll into view. When I use it, it doesn't scroll to the recommend section unless I use a timeout which seems odd.

Comment (zaguiini) [reply client/hosting/performance/site-performance.tsx:146]: Alright 👍🏻

Comment (zaguiini) [reply client/hosting/performance/site-performance.tsx:146]: Oh, actually, you can try passing the full URL, like we're doing in the `PageSelector` prop.

Comment (vykes-mac) [reply client/hosting/performance/site-performance.tsx:146]: The page.replace totally invalidates the scroll into view. I'm going to merge as is for now

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/16874681

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @vykes-mac for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16874681).

**Pull request merged**